src/validators.ts

```diff
@@ -6,9 +6,18 @@ import { CurrentEpochValidatorInfo, NextEpochValidatorInfo } from './providers/p
 /** Finds seat price given validators stakes and number of seats.
  *  Calculation follow the spec: https://nomicon.io/Economics/README.html#validator-selection
  * @params validators: current or next epoch validators.
- * @params numSeats: number of seats.
+ * @params maxNumberOfSeats: maximum number of seats in the network.
+ * @params minimumStakeRatio: minimum stake ratio 
+ * @params protocolVersion: version of the protocol from genesis config
  */
-export function findSeatPrice(validators: (CurrentEpochValidatorInfo | NextEpochValidatorInfo)[], numSeats: number): BN {
+export function findSeatPrice(validators: (CurrentEpochValidatorInfo | NextEpochValidatorInfo)[], maxNumberOfSeats: number, minimumStakeRatio?: number, protocolVersion?: number): BN {
+    if (!minimumStakeRatio || !protocolVersion || protocolVersion < 49) {
+        return findSeatPriceForProtocolBefore49(validators, maxNumberOfSeats);
+    }
+    return findSeatPriceForProtocolAfter49(validators, maxNumberOfSeats, minimumStakeRatio);
```

Comment:
I feel the default should actually be "After49" version. I feel that given that one will only receive a new version near-api-js after an explicit upgrade, it is better to throw an exception that the old version of the function does not work without the extra arguments rather than letting developers down with the wrong seat price number.

P.S. We need to have a reference where developers can read more about the change of the seat allocation algorithm. @nikurt Is there any post/changelog about the new seat allocation?

Reply:
@frol we do not know `minimumStakeRatio` to use a new function as default.
We can throw deprecation warning like this and hardcode a value:
```javascript
export function findSeatPrice(validators: (CurrentEpochValidatorInfo | NextEpochValidatorInfo)[], maxNumberOfSeats: number, minimumStakeRatio?: number, protocolVersion?: number): BN {
    if (protocolVersion && protocolVersion < 49) {
        return findSeatPriceForProtocolBefore49(validators, maxNumberOfSeats);
    }
    if (!minimumStakeRatio) {
        //TODO: deprecation warning
        minimumStakeRatio =  6250;
    }
    return findSeatPriceForProtocolAfter49(validators, maxNumberOfSeats, minimumStakeRatio);
}
```
Or we can change the function signature and make a breaking change like this:
```javascript
export function findSeatPrice(validators: (CurrentEpochValidatorInfo | NextEpochValidatorInfo)[], maxNumberOfSeats: number, minimumStakeRatio: number, protocolVersion?: number): BN {
    if (protocolVersion && protocolVersion < 49) {
        return findSeatPriceForProtocolBefore49(validators, maxNumberOfSeats);
    }
    return findSeatPriceForProtocolAfter49(validators, maxNumberOfSeats, minimumStakeRatio);
}
```
I like the second option better.

Reply:
I would mix the two to make sure that if someone uses near-api-js from JS, they don't get garbage in return.

Reply:
Ok, merged.